api/lib/infrastructure/repositories/url-repository.js

```diff
@@ -87,6 +88,32 @@ async function addSheetToGoogleSheet(dataToUpload, sheetName, spreadsheetId) {
   }
 }
 
+async function clearOlderSheets(spreadsheetId) {
+  try {
+    const auth = await getAuthToken(config.googleAuthCredentials);
+    const spreadsheet = await sheets.spreadsheets.get({ spreadsheetId, auth });
+    const limitDate = new Date();
+    limitDate.setMonth(new Date().getMonth() - DELETE_EXTERNAL_URL_SHEET_DELAY_MONTHS);
+    const isSheetOlderThanLimitDate = function(sheetTitle, limitDate) {
+      const standardFormattedDate = sheetTitle.split('/').reverse().join('-');
+      const sheetDate = new Date(standardFormattedDate);
+      if (isNaN(sheetDate)) {
+        return false;
+      }
+      return sheetDate.getTime() < limitDate.getTime();
+    };
+    const sheetsToDelete = spreadsheet.data.sheets.filter((sheet) => isSheetOlderThanLimitDate(sheet.properties.title, limitDate));
+    for (const sheetToDelete of sheetsToDelete) {
+      await sheets.spreadsheets.batchUpdate({
+        spreadsheetId,
+        auth,
+        resource: { requests: [ { deleteSheet: { sheetId: sheetToDelete.properties.sheetId } }] } });
```

Comment:
On pourrait peut-être enlever la boucle for et faire un vrai batchUpdate

```suggestion
        resource: { 
          requests: sheetsToDelete.map((sheetToDelete) => {
            return { deleteSheet: { sheetId: sheetToDelete.properties.sheetId } };
          }),
        },
      });
```

Reply:
je vais essayer